payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,116 @@
+package com.stripe.android.payments.core.authentication
+
+import androidx.activity.result.ActivityResultLauncher
+import com.stripe.android.Logger
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentBrowserAuthStarter
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.payments.core.injection.AuthenticationModule
+import com.stripe.android.payments.core.injection.DaggerAuthenticationComponent
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.view.AuthActivityStarter
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Default registry to provide look ups for [IntentAuthenticator].
+ * Should be only accessed through [DefaultIntentAuthenticatorRegistry.getInstance].
+ */
+internal class DefaultIntentAuthenticatorRegistry @Inject internal constructor() :
+    IntentAuthenticatorRegistry {
+
+    // default authenticators always packaged with the SDK
+    @Inject
+    lateinit var webIntentAuthenticator: WebIntentAuthenticator
+
+    @Inject
+    lateinit var noOpIntentAuthenticator: NoOpIntentAuthenticator
+
+    // authenticators requiring a 3p SDK
+    // TODO(ccen) move them to a dedicated module
+    @Inject
+    lateinit var stripe3DS2Authenticator: Stripe3DS2Authenticator
+
+    override fun lookUp(stripeIntent: StripeIntent): IntentAuthenticator {
+        if (!stripeIntent.requiresAction()) {
+            return noOpIntentAuthenticator
+        }
+
+        when (val nextActionData = stripeIntent.nextActionData) {
+            is StripeIntent.NextActionData.SdkData.Use3DS2 -> {
+                return stripe3DS2Authenticator
+            }
+            is StripeIntent.NextActionData.SdkData.Use3DS1 -> {
+                // can only triggered when `use_stripe_sdk=true`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.RedirectToUrl -> {
+                // can only triggered when `use_stripe_sdk=false`
```

Comment:
In what cases is use_stripe_sdk false?

Reply:
This is [a API parameter](https://stripe.com/docs/api/payment_intents/create#create_payment_intent-use_stripe_sdk) set by useres, these checks are preserved from [StripePaymentCongtroller#handleNextAction](https://github.com/stripe/stripe-android/blob/ef2197fabeb47436a4119e23dab74041c31320d4/payments-core/src/main/java/com/stripe/android/StripePaymentController.kt#L533)